tsconfig.json

```diff
@@ -6,7 +6,7 @@
     "allowJs": true,
     "checkJs": true,
     "target": "ES2022",
-    "module": "ES2022",
```

Comment:
This change inspired by
https://github.com/arethetypeswrong/arethetypeswrong.github.io/blob/64b1f489736bd48ebb4e3b72fb078b44437d59e0/docs/problems/InternalResolutionError.md
> This problem can occur with tsc alone if the library author compiles with --moduleResolution node (also known as node10) or bundler, especially in combination with a package.json that has "type": "module"

and

> Library authors should always compile/check with --module node16 --moduleResolution node16 or nodenext (note that module im